libsolidity/interface/ABI.cpp

```diff
@@ -40,7 +40,7 @@ bool anyDataStoredInStorage(TypePointers const& _pointers)
 
 Json::Value ABI::generate(ContractDefinition const& _contractDef)
 {
-	Json::Value abi(Json::arrayValue);
+	vector<Json::Value> abi;
```

Comment:
```suggestion
	auto compare = [](Json::Value const& a, Json::Value const& b) { return make_tuple(_a["type"], _a["name"]) < make_tuple(_b["type"], _b["name"]); };
	multiset<Json::Value, decltype(compare)> abi(compare);
```
and then just ``emplace``ing everything should be the same without the need to sort in the end, right?